ReactNativeClient/lib/services/InteropService_Importer_Md.js

```diff
@@ -54,21 +58,50 @@ class InteropService_Importer_Md extends InteropService_Importer_Base {
 		}
 	}
 
+	/**
+	 * Parse text for links, attempt to find local file, if found create Joplin resource
+	 * and update link accordingly.
+	 *
+	 * mdImageTagRegex matches markdown image tags.
+	 * !\[ and \] matches start of tag containing alt text.
+	 * Within this the [^\]]* matches multiple characters not in this set;
+	 * in this case the set is "]" - so match anything that is not this char.
+	 * We need this to avoid overshooting if another "]" follows after the
+	 * match.
+	 * \(.*?\) to match the part of the tag containing the link.
+	 */
+	async importLocalImages(filePath, md) {
+		const mdImageTagRegex = /!\[[^\]]*\]\((.*?)\)/g;
+		let updated = md;
+		let match;
+		while ((match = mdImageTagRegex.exec(md)) !== null) {
+			const attachmentPath = path.resolve(path.dirname(filePath), match[1]);
```

Comment:
Please use `dirname` in `path-utils`. In general we avoid importing runtime-specific packages because the code needs to work in React Native too (actually this one doesn't, but it's better to keep things consistent in case we decide to make it work in RN).

Reply:
@laurent22 Thanks for the info about avoiding runtime-specific deps, makes sense.
I'm having difficulty figuring out how to achieve path resolution using the helpers in `path-utils` - unless I'm missing something `dirname` wont resolve the path for me.
Consider these logs of different values from the tests:

```
OK resolve: path.resolve(path.dirname(filePath), match[1]):  /Users/BB/projects/joplin/CliClient/tests-build/support/photo.jpg
KO dirname: dirname(dirname(filePath) + '/' + match[1]) + '.' + fileExtension(match[1]) /Users/BB/projects/joplin/CliClient/tests-build/md_to_md/../support.jpg
OK resolve: path.resolve(path.dirname(filePath), match[1]):  /Users/BB/projects/joplin/CliClient/tests-build/support/photo.jpg
KO dirname: dirname(dirname(filePath) + '/' + match[1]) + '.' + fileExtension(match[1]) /Users/BB/projects/joplin/CliClient/tests-build/md_to_md/../support.jpg
OK resolve: path.resolve(path.dirname(filePath), match[1]):  /Users/BB/projects/joplin/CliClient/tests-build/md_to_md/does_not_exist.png
KO dirname: dirname(dirname(filePath) + '/' + match[1]) + '.' + fileExtension(match[1]) /Users/BB/projects/joplin/CliClient/tests-build/md_to_md.png
OK resolve: path.resolve(path.dirname(filePath), match[1]):  /Users/BB/projects/joplin/CliClient/tests-build/support
KO dirname: dirname(dirname(filePath) + '/' + match[1]) + '.' + fileExtension(match[1]) /Users/BB/projects/joplin/CliClient/tests-build/md_to_md/.../support
```
Any suggestions?

Reply:
think I've got a working combo of methods from `path-utils` now 🙂

Reply:
@laurent22 Was able to get a usable path with a combo of methods from `path-utils` (see changes) but it's pretty ugly and involves hardcoding a `/` - not sure if that will be a problem?
Any better suggestions?

Reply:
This is fine for now. In general it's harmless to hard-code `/` since all libraries support this. If I'm not mistaken the only place it's not hardcoded is in the dev scripts that run DOS commands.